GitCommands/Git/RevisionDiffProvider.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Diagnostics;
+
+namespace GitCommands
+{
+    /// <summary>
+    /// Translate GitRevision including artificial commits to diff options
+    /// Closely related to GitRevision.cs 
+    /// </summary>
+    public static class RevisionDiffProvider
+    {
+        private const string StagedOpt = "--cached";
+
+        /// <summary>
+        /// Translate the revision string to an option string
+        /// Artificial "commits" are options, handle aliases too
+        /// (order and handling of empty arguments is not handled here)
+        /// </summary>
+        /// <param name="rev"></param>
+        /// <returns></returns>
+        private static string ArtificialToDiffOptions(string rev)
+        {
+            if (rev.IsNullOrEmpty() || rev == GitRevision.UnstagedGuid) { rev = string.Empty; }
+            else if (rev == "^" || rev == GitRevision.UnstagedGuid + "^" || rev == GitRevision.IndexGuid) { rev = StagedOpt; }
+            else
+            {
+                //Normal commit
+                if (rev == "^^" || rev == GitRevision.UnstagedGuid + "^^" || rev == GitRevision.IndexGuid + "^") { rev = "HEAD"; }
+                else if (rev == "^^^" || rev == GitRevision.UnstagedGuid + "^^^" || rev == GitRevision.IndexGuid + "^^") { rev = "HEAD^"; }
+                rev = rev.QuoteNE();
+            }
+
+            return rev;
+        }
+
+        /// <summary>
+        /// options to git-diff from GE arguments, including artificial commits
+        /// </summary>
+        /// <param name="from">The first revision</param>
+        /// <param name="to">The second "current" revision</param>
+        /// <returns></returns>
+        public static string Get(string from, string to)
```

Comment:
Is it necessary to validate the the input parameters? E.g. are null or empty string acceptable for either of them?
If not - then throw `ArgumentNullException` or `ArgumentException`

Reply:
null and empty is valid and used by the code (same as unstaged)